catalyst_voices/packages/internal/catalyst_voices_models/lib/src/proposal/proposal.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:catalyst_cardano_serialization/catalyst_cardano_serialization.dart';
+import 'package:equatable/equatable.dart';
+
+// Note. This enum may be deleted later. Its here for backwards compatibility.
+enum ProposalStatus { ready, draft, inProgress, private, open, live, completed }
+
+enum ProposalPublish { draft, published }
+
+enum ProposalAccess { private, public }
+
+final class Proposal extends Equatable {
+  final String id;
+  final String title;
+  final String description;
+  final DateTime updateDate;
+  final DateTime? fundedDate;
+  final Coin fundsRequested;
+  final ProposalPublish publish;
+  final ProposalAccess access;
+
+  // This may be a reference to class
+  final String category;
+
+  // Those may be getters.
+  final int commentsCount;
+  final int completedSegments;
+  final int totalSegments;
+
+  const Proposal({
+    required this.id,
+    required this.title,
+    required this.description,
+    required this.updateDate,
+    this.fundedDate,
+    required this.fundsRequested,
+    required this.publish,
+    required this.access,
+    required this.category,
+    required this.commentsCount,
+    required this.completedSegments,
+    required this.totalSegments,
+  });
+
+  bool get isFunded => fundedDate != null;
+
+  // Note. Not yet sure how it should be computed
```

Comment:
It's a field, user needs to create a proposal, then it's a draft. User submits a proposal then it's published. When a fund starts the proposal status will change as well. In short, backend should manage it

Reply:
done